app/src/main/java/de/xikolo/controllers/helper/DownloadViewHelper.kt

```diff
@@ -159,21 +161,13 @@ class DownloadViewHelper(
         }
 
         buttonOpenDownload.setOnClickListener {
-            val file = downloadManager.getDownloadFile(downloadAsset)
-            if (file != null) {
-                val target = Intent(Intent.ACTION_VIEW)
-                target.setDataAndType(FileProviderUtil.getUriForFile(file), downloadAsset.mimeType)
-                target.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
-                target.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
-
-                val intent = Intent.createChooser(target, null)
-                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            download.openIntent?.let { intent ->
                 try {
                     App.instance.startActivity(intent)
                 } catch (e: ActivityNotFoundException) {
                     activity.showToast(R.string.toast_no_file_viewer_found)
                 }
-            } else {
+            } ?: run {
```

Comment:
The block is only executed when `download.openIntent` is `null`. Is it that what you mean?
Another approach would be to hide the "Open" button when the `openIntent` is `null`

Reply:
I was talking about the usage of the programming syntax:

`if (download.openIntent != null) { ... } else { ... }`
vs
`download.openIntent?.let { intent -> ... } ?: run { ... }`

Usually, I would prefer the first option. The second one unwraps the nullable intent, but that should do the first one automatically as well since the compiler remembers the check, or?

Reply:
Yes you are right